src/FlashList.tsx

```diff
@@ -0,0 +1,121 @@
+import { FlashList as SPFlashList, FlashListProps } from '@shopify/flash-list'
+import React from 'react'
+
+import { AnimatedFlashList } from './helpers'
+import {
+  useAfterMountEffect,
+  useChainCallback,
+  useCollapsibleStyle,
+  useConvertAnimatedToValue,
+  useScrollHandlerY,
+  useSharedAnimatedRef,
+  useTabNameContext,
+  useTabsContext,
+  useUpdateScrollViewContentSize,
+} from './hooks'
+
```

Comment:
// Load FlashList dynamically or print a friendly error message
let SPFlashList, FlashListProps
try {
  const flashListModule = require('@shopify/flash-list')
  SPFlashList = flashListModule.FlashList
  FlashListProps = flashListModule.FlashListProps
} catch (error) {
  if (error.code === 'MODULE_NOT_FOUND') {
    console.error(
      'The optional dependency @shopify/flash-list is not installed. Please install it to use the FlashList component.'
    )
  } else {
    throw error
  }
}

Reply:
Need also to forward the typescript types of flash-list properly here.